### PR TITLE
fix namespace resource filtering

### PR DESCRIPTION
**Pull request by deads2k**
namespace as a resource didn't work right.

### CODE REVIEW

File: pkg/cmd/audit/audit_filter.go [hunk 1]
```diff
@@ -222,13 +222,30 @@ func URIToParts(uri string) (string, schema.GroupVersionResource, string) {
...
 				name = parts[3]
 				return ns, gvr, name
 			}
+		case len(parts) == 3 && parts[2] == "namespaces" : // a namespace request /api/v1/namespaces
+			gvr.Resource = parts[2]
+			return "", gvr, ""
+
+		case len(parts) == 4 && parts[2] == "namespaces", // a namespace request /api/v1/namespaces/<name>
+			len(parts) == 5 && parts[2] == "namespaces" && parts[2] == "finalize", // a namespace request /api/v1/namespaces/<name>/finalize
+			len(parts) == 5 && parts[2] == "namespaces" && parts[2] == "finalize": // a namespace request /api/v1/namespaces/<name>/status
```
Comment (retroflexer) [inline pkg/cmd/audit/audit_filter.go:238]: These two lines look identical, although the comments are different. 

Furthermore, the "finalize" should be compared against parts[4], isn't it? 

From the comments it appears you probably want to do this:
```
len(parts) == 5 && parts[2] == "namespaces" && parts[4] == "finalize", // a namespace request /api/v1/namespaces/<name>/finalize
			len(parts) == 5 && parts[2] == "namespaces" && parts[4] == "status": // a namespace request /api/v1/namespaces/<name>/status 
```

Comment (deads2k) [reply pkg/cmd/audit/audit_filter.go:238]: awkward. Thanks

File: pkg/cmd/audit/audit_filter.go [hunk 2]
```diff
@@ -222,13 +222,30 @@ func URIToParts(uri string) (string, schema.GroupVersionResource, string) {
 			return ns, gvr, name
 		}
 
-		if parts[2] != "namespaces" {
+		switch{
```
Comment (tkashem) [inline pkg/cmd/audit/audit_filter.go:225]: just curious whether client-go or apimachinery already has a function (covered by unit tests) that deconstructs a given url.

Comment (deads2k) [reply pkg/cmd/audit/audit_filter.go:225]: > just curious whether client-go or apimachinery already has a function (covered by unit tests) that deconstructs a given url.

it's in k8s.io/apiserver I think.

Comment (tkashem) [reply pkg/cmd/audit/audit_filter.go:225]: what we have now works; as a future goal I am happy to take this as a TODO for me to utilize the function in `k8s.io/apiserver`.

**Pull request merged**